src/pke/lib/schemerns/rns-leveledshe.cpp

```diff
@@ -537,4 +537,22 @@ void LeveledSHERNS::AdjustForMultInPlace(Ciphertext<DCRTPoly>& ciphertext1, Ciph
     }
 }
 
+Ciphertext<DCRTPoly> LeveledSHERNS::ComposedEvalMult(ConstCiphertext<DCRTPoly> ciphertext1,
+                                                     ConstCiphertext<DCRTPoly> ciphertext2,
+                                                     const EvalKey<DCRTPoly> evalKey) const {
+    const auto cc           = ciphertext1->GetCryptoContext();
+    auto algo               = ciphertext1->GetCryptoContext()->GetScheme();
+    const auto cryptoParams = std::dynamic_pointer_cast<CryptoParametersRNS>(ciphertext1->GetCryptoParameters());
+    Ciphertext<DCRTPoly> ciphertext = EvalMult(ciphertext1, ciphertext2);
+    algo->KeySwitchInPlace(ciphertext, evalKey);
+    uint32_t levelsToDrop = BASE_NUM_LEVELS_TO_DROP;
+    if (cryptoParams->GetScalingTechnique() == COMPOSITESCALINGAUTO ||
+        cryptoParams->GetScalingTechnique() == COMPOSITESCALINGMANUAL) {
+        const auto cryptoRNSParams = std::dynamic_pointer_cast<CryptoParametersRNS>(cryptoParams);
```

Comment:
cryptoRNSParams is not needed, you could re-use cryptoParams

Reply:
Not sure what happened here. Also just above, the line `const auto cc           = ciphertext1->GetCryptoContext();` is not needed either.